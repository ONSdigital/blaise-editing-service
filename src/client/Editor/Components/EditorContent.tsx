import { ONSPanel, ONSSelect, ONSTable } from 'blaise-design-system-react-components';
import { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import { EditorInformation } from '../../../common/interfaces/editorInterface';
import { QuestionnaireDetails } from '../../../common/interfaces/surveyInterface';

interface EditorsContentProps {
  editorInformation: EditorInformation;
  questionnaire: QuestionnaireDetails
}

export default function EditorContent({ editorInformation, questionnaire }: EditorsContentProps): ReactElement {
  return (
    <div className="editorContent" data-testid={`${questionnaire.questionnaireName}-editorContent`}>
      <ONSPanel status="info">
        <dl
          className="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
          title="editorContent"
          data-testid="editorContent-dl"
          style={{ padding: '0 0 15px 5px' }}
        >
          <dt className="ons-description-list__term ons-grid__col ons-col-5@m">Field period:</dt>
          <dd className="ons-description-list__value ons-grid__col ons-col-7@m"><strong>{questionnaire.fieldPeriod}</strong></dd>
          <dt className="ons-description-list__term ons-grid__col ons-col-5@m">Cases assigned to me:</dt>
          <dd className="ons-description-list__value ons-grid__col ons-col-7@m"><strong>{editorInformation.numberOfCasesAllocated}</strong></dd>

        </dl>
      </ONSPanel>
      <ONSSelect
        defaultValue="in-progress"
        id="filter-cases"
        label="Filter cases"
        onChange={() => {}}
        options={[
          {
            label: 'In Progress',
            value: 'in-progress',
          },
          {
            label: 'Complete',
            value: 'complete',
          },
          {
            label: 'Not Started',
            value: 'not-started',
          },
        ]}
        value=""
      />

      <ONSTable
        columns={[
          'Case ID',
          'Status',
          '',
        ]}
        tableID={`${questionnaire.questionnaireName}-Case-table`}
      >
        <>
          {editorInformation.Cases.map((caseDetails) => (
            <tr
              className="ons-table__row"
              key={caseDetails.CaseId}
            >
              <td className="ons-table__cell" aria-label={`${questionnaire.questionnaireName}-CaseID`}>
                {caseDetails.CaseId}
              </td>
              <td className="ons-table__cell status" aria-label={`${questionnaire.questionnaireName}-EditStatus`}>
                {caseDetails.EditStatus}
              </td>
              <td className="ons-table__cell links">
                <Link className="Edit" to="/summary">Summary</Link>
                {' | '}
                <Link className="Edit" to="/">Edit</Link>
              </td>
            </tr>
          ))}
        </>
      </ONSTable>

    </div>
  );
}
