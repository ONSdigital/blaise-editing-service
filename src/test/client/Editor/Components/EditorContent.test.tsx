import { act, render, RenderResult } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import EditorContent from '../../../../client/Editor/Components/EditorContent';
import { questionnaireDetails2MockObject } from '../../../mockObjects/questionnaireListMockObject';
import { EditorInformationMockObject } from '../../MockObjects/EditorMockObjects';

const editorInformation = EditorInformationMockObject;
const QuestionnaireDetails = questionnaireDetails2MockObject;
let view:RenderResult;

describe('Given there is a survey and cases avalible for an editor', () => {
  it('should display the editor page correctly', async () => {
    // arrange
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <EditorContent editorInformation={editorInformation} questionnaire={QuestionnaireDetails} />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });

  it('should render the editor page correctly', async () => {
    // arrange
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <EditorContent editorInformation={editorInformation} questionnaire={QuestionnaireDetails} />
        </BrowserRouter>,
      );
    });

    // assert

    const expectedView = view.getByTestId(`${questionnaireDetails2MockObject.questionnaireName}-editorContent`);
    expect(expectedView).toHaveTextContent(questionnaireDetails2MockObject.fieldPeriod ?? 'Not Found');
    expect(expectedView).toHaveTextContent(EditorInformationMockObject.numberOfCasesAllocated.toString());
    EditorInformationMockObject.Cases.forEach((caseDetails) => {
      expect(expectedView).toHaveTextContent(caseDetails.CaseId);
      expect(expectedView).toHaveTextContent(caseDetails.EditStatus.toString());
    });
  });
});
