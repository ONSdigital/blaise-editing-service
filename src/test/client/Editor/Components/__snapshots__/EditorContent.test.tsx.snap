// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given there is a survey and cases avalible for an editor should display the editor page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editorContent"
        data-testid="FRS2504A-editorContent"
      >
        <div
          class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            <dl
              class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
              data-testid="editorContent-dl"
              style="padding: 0px 0px 15px 5px;"
              title="editorContent"
            >
              <dt
                class="ons-description-list__term ons-grid__col ons-col-5@m"
              >
                Field period:
              </dt>
              <dd
                class="ons-description-list__value ons-grid__col ons-col-7@m"
              >
                <strong>
                  January 2021
                </strong>
              </dd>
              <dt
                class="ons-description-list__term ons-grid__col ons-col-5@m"
              >
                Cases assigned to me:
              </dt>
              <dd
                class="ons-description-list__value ons-grid__col ons-col-7@m"
              >
                <strong>
                  5
                </strong>
              </dd>
            </dl>
          </div>
        </div>
        <div>
          <label
            class="ons-label"
            for="filter-cases"
          >
            Filter cases
             
          </label>
          <select
            class="ons-input "
            id="filter-cases"
            name="select"
          >
            <option
              data-testid="select-filter-cases"
              disabled=""
              value=""
            >
              Select an option
            </option>
            <option
              data-testid="option-filter-cases-in-progress"
              selected=""
              value="in-progress"
            >
              In Progress
            </option>
            <option
              data-testid="option-filter-cases-complete"
              value="complete"
            >
              Complete
            </option>
            <option
              data-testid="option-filter-cases-not-started"
              value="not-started"
            >
              Not Started
            </option>
          </select>
        </div>
        <table
          class="ons-table"
        >
          <thead
            class="ons-table__head"
          >
            <tr
              class="ons-table__row"
            >
              <th
                class="ons-table__header"
                scope="col"
              >
                Case ID
              </th>
              <th
                class="ons-table__header"
                scope="col"
              >
                Status
              </th>
              <th
                class="ons-table__header"
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="ons-table__body"
          >
            <tr
              class="ons-table__row"
              data-testid="case-table-row-10001011"
            >
              <td
                class="ons-table__cell"
              >
                10001011
              </td>
              <td
                class="ons-table__cell status"
              >
                3
              </td>
              <td
                class="ons-table__cell links"
              >
                <a
                  class="Edit"
                  href="/summary"
                >
                  Summary
                </a>
                 | 
                <a
                  class="Edit"
                  href="/"
                >
                  Edit
                </a>
              </td>
            </tr>
            <tr
              class="ons-table__row"
              data-testid="case-table-row-10001012"
            >
              <td
                class="ons-table__cell"
              >
                10001012
              </td>
              <td
                class="ons-table__cell status"
              >
                1
              </td>
              <td
                class="ons-table__cell links"
              >
                <a
                  class="Edit"
                  href="/summary"
                >
                  Summary
                </a>
                 | 
                <a
                  class="Edit"
                  href="/"
                >
                  Edit
                </a>
              </td>
            </tr>
            <tr
              class="ons-table__row"
              data-testid="case-table-row-10001013"
            >
              <td
                class="ons-table__cell"
              >
                10001013
              </td>
              <td
                class="ons-table__cell status"
              >
                0
              </td>
              <td
                class="ons-table__cell links"
              >
                <a
                  class="Edit"
                  href="/summary"
                >
                  Summary
                </a>
                 | 
                <a
                  class="Edit"
                  href="/"
                >
                  Edit
                </a>
              </td>
            </tr>
            <tr
              class="ons-table__row"
              data-testid="case-table-row-10001014"
            >
              <td
                class="ons-table__cell"
              >
                10001014
              </td>
              <td
                class="ons-table__cell status"
              >
                3
              </td>
              <td
                class="ons-table__cell links"
              >
                <a
                  class="Edit"
                  href="/summary"
                >
                  Summary
                </a>
                 | 
                <a
                  class="Edit"
                  href="/"
                >
                  Edit
                </a>
              </td>
            </tr>
            <tr
              class="ons-table__row"
              data-testid="case-table-row-10001015"
            >
              <td
                class="ons-table__cell"
              >
                10001015
              </td>
              <td
                class="ons-table__cell status"
              >
                3
              </td>
              <td
                class="ons-table__cell links"
              >
                <a
                  class="Edit"
                  href="/summary"
                >
                  Summary
                </a>
                 | 
                <a
                  class="Edit"
                  href="/"
                >
                  Edit
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editorContent"
      data-testid="FRS2504A-editorContent"
    >
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          <dl
            class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
            data-testid="editorContent-dl"
            style="padding: 0px 0px 15px 5px;"
            title="editorContent"
          >
            <dt
              class="ons-description-list__term ons-grid__col ons-col-5@m"
            >
              Field period:
            </dt>
            <dd
              class="ons-description-list__value ons-grid__col ons-col-7@m"
            >
              <strong>
                January 2021
              </strong>
            </dd>
            <dt
              class="ons-description-list__term ons-grid__col ons-col-5@m"
            >
              Cases assigned to me:
            </dt>
            <dd
              class="ons-description-list__value ons-grid__col ons-col-7@m"
            >
              <strong>
                5
              </strong>
            </dd>
          </dl>
        </div>
      </div>
      <div>
        <label
          class="ons-label"
          for="filter-cases"
        >
          Filter cases
           
        </label>
        <select
          class="ons-input "
          id="filter-cases"
          name="select"
        >
          <option
            data-testid="select-filter-cases"
            disabled=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-filter-cases-in-progress"
            selected=""
            value="in-progress"
          >
            In Progress
          </option>
          <option
            data-testid="option-filter-cases-complete"
            value="complete"
          >
            Complete
          </option>
          <option
            data-testid="option-filter-cases-not-started"
            value="not-started"
          >
            Not Started
          </option>
        </select>
      </div>
      <table
        class="ons-table"
      >
        <thead
          class="ons-table__head"
        >
          <tr
            class="ons-table__row"
          >
            <th
              class="ons-table__header"
              scope="col"
            >
              Case ID
            </th>
            <th
              class="ons-table__header"
              scope="col"
            >
              Status
            </th>
            <th
              class="ons-table__header"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="ons-table__body"
        >
          <tr
            class="ons-table__row"
            data-testid="case-table-row-10001011"
          >
            <td
              class="ons-table__cell"
            >
              10001011
            </td>
            <td
              class="ons-table__cell status"
            >
              3
            </td>
            <td
              class="ons-table__cell links"
            >
              <a
                class="Edit"
                href="/summary"
              >
                Summary
              </a>
               | 
              <a
                class="Edit"
                href="/"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="ons-table__row"
            data-testid="case-table-row-10001012"
          >
            <td
              class="ons-table__cell"
            >
              10001012
            </td>
            <td
              class="ons-table__cell status"
            >
              1
            </td>
            <td
              class="ons-table__cell links"
            >
              <a
                class="Edit"
                href="/summary"
              >
                Summary
              </a>
               | 
              <a
                class="Edit"
                href="/"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="ons-table__row"
            data-testid="case-table-row-10001013"
          >
            <td
              class="ons-table__cell"
            >
              10001013
            </td>
            <td
              class="ons-table__cell status"
            >
              0
            </td>
            <td
              class="ons-table__cell links"
            >
              <a
                class="Edit"
                href="/summary"
              >
                Summary
              </a>
               | 
              <a
                class="Edit"
                href="/"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="ons-table__row"
            data-testid="case-table-row-10001014"
          >
            <td
              class="ons-table__cell"
            >
              10001014
            </td>
            <td
              class="ons-table__cell status"
            >
              3
            </td>
            <td
              class="ons-table__cell links"
            >
              <a
                class="Edit"
                href="/summary"
              >
                Summary
              </a>
               | 
              <a
                class="Edit"
                href="/"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="ons-table__row"
            data-testid="case-table-row-10001015"
          >
            <td
              class="ons-table__cell"
            >
              10001015
            </td>
            <td
              class="ons-table__cell status"
            >
              3
            </td>
            <td
              class="ons-table__cell links"
            >
              <a
                class="Edit"
                href="/summary"
              >
                Summary
              </a>
               | 
              <a
                class="Edit"
                href="/"
              >
                Edit
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
